cmake/ElektraCompiling.cmake

```diff
@@ -61,10 +61,27 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
 	if (ENABLE_DEBUG)
 		set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=undefined -fsanitize=integer")
-		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=undefined -fsanitize=integer")
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lubsan")
 	endif()
 endif()
 
+if (ENABLE_ASAN)
+	set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+	set (ASAN_LIBRARY "-lasan") #this is needed for GIR to put asan in front
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+		# this is currently a workaround so gtests compile with ASAN
+		#TODO someone with expertise migrate this into LibAddTest
+		set (COMMON_FLAGS "${COMMON_FLAGS} -lpthread")
```

Comment:
So you say that it actually is *not* needed globally but only gtest suddenly does not link with ENABLE_ASAN? Then we should not enable it globally, but fix the linking order for the gtests.

We had a issue with pthread and gtest already here: https://github.com/ElektraInitiative/libelektra/issues/737

You can either find out the exact dependencies of every library and do a topological sort or you can simply guess and shuffle around the linker flags until it works (as done in #737)